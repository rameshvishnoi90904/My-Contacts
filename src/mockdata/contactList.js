export const contactList = [
    {
      recordID: '1',
      givenName: 'Elon',
      familyName: 'Musk',
      emailAddresses: [{
        email: 'andre@gmail.com', 
        label: 'work'}],
      phoneNumbers: [{
        number: '+23493023902',
        label: 'personal'}],
      picture: '',
    },
    {
      recordID: '2',
      givenName: 'Jack',
      familyName: 'Ma',
      emailAddresses: [{
        email: 'andre@gmail.com', 
        label: 'work'}],
      phoneNumbers: [{
        number: '+23493023902',
        label: 'personal'}],
      picture: '',
    },
    {
      recordID: '3',
      givenName: 'Larry',
      familyName: 'Page',
      emailAddresses: [{
        email: 'andre@gmail.com', 
        label: 'work'}],
      phoneNumbers: [{
        number: '+23493023902',
        label: 'personal'}],
      picture: '',
    },
    {
      recordID: '4',
      givenName: 'Mark',
      familyName: 'Zuck',
      emailAddresses: [{
        email: 'andre@gmail.com', 
        label: 'work'}],
      phoneNumbers: [{
        number: '+23493023902',
        label: 'personal'}],
      picture: '',
    },
    {
      recordID: '5',
      givenName: 'Jeff',
      familyName: 'Bezos',
      emailAddresses: [{
        email: 'andre@gmail.com', 
        label: 'work'}],
      phoneNumbers: [{
        number: '+23493023902',
        label: 'personal'}],
      picture: '',
    },
];