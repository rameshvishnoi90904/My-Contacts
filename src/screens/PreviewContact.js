import React from 'react';
import {
    Button,
  ScrollView,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import { getUserInitials } from '../helper';
import Icon from 'react-native-vector-icons/FontAwesome5'
import EIcon from 'react-native-vector-icons/Entypo'

const PreviewContact = ({navigation, route}) => {
    const contactInfo = route.params.contactInfo;
    const selectedColor = route.params.selectedColor;
    
    const pList = contactInfo.phoneNumbers.map((item) => {
        return (
            <View style={styles.phoneNumberItem} key={item.number}>
                <View style={{flex: 1}}>
                    <Text style={styles.numberText}>{item.number}</Text>
                    <Text style={styles.numberTypeLabel}>{item.label}</Text>
                </View>
                <Icon name="phone-alt" size={22} style={{marginHorizontal: 16}}/>
            </View>
        )
    });
    const eList = contactInfo.emailAddresses.map((item) => {
        return (
            <View style={styles.phoneNumberItem} key={item.email}>
                <View style={{flex:1}}>
                    <Text style={styles.emailText}>{item.email}</Text>
                    <Text style={styles.numberTypeLabel}>{item.label}</Text>
                </View>
                <EIcon name="email" size={22} style={{marginHorizontal: 16}}/>
            </View>
        )
    });

    const navigateToEdit = () => {
        navigation.navigate("Edit Contact",{
            contactInfo: contactInfo
        });
    }
    return (
        <ScrollView style={styles.container}>
            <View style={styles.profilePictureContainer}>
                <View style={[styles.profilePicture, {backgroundColor: selectedColor}]}>
                    <Text style={styles.userInitials}>{getUserInitials(contactInfo)}</Text>
                </View>
            </View>
            <Text 
                accessibilityRole="header"
                testID="contact-name"
                style={styles.contactName}>{contactInfo.givenName} {contactInfo.familyName}</Text>
            <View style={styles.phoneNumberContainer}>
                {pList}
            </View>
            <View style={styles.phoneNumberContainer}>
                {eList}
            </View>
            <View style={{flex: 1}}></View>
            <View style={{marginBottom: 16, marginHorizontal: 16}}>
                <Button 
                title="Edit"
                accessibilityLabel="Edit contact"
                accessibilityHint="Opens Contact detail in new page for editing"
                onPress={navigateToEdit}/>
            </View>
        </ScrollView>
    )
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: 'white'
    },
    profilePicture: {
        width: 120,
        height: 120, 
        borderRadius: 60,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'gray'
    },
    profilePictureContainer: {
        alignItems: 'center',
        marginTop: 32,
        marginBottom: 16
    },
    contactName: {
        fontSize: 22,
        textAlign: 'center'
    },
    topSection: {
        paddingBottom: 32
    },
    phoneNumberItem: {
        borderBottomWidth: 1,
        paddingBottom: 12,
        paddingTop: 8,
        borderBottomColor: '#C9C9C9',
        flexDirection: 'row',
        alignItems: 'center'
    },
    numberText: {
        fontSize: 20,
    },
    emailText: {
        fontSize: 16,
    },
    phoneNumberContainer: {
        paddingHorizontal: 16,
        marginVertical: 8
    },
    numberTypeLabel: {
        textTransform: 'capitalize',
        color: '#6D6D6D',
        fontSize: 12
    },
    userInitials: {
        color: 'white',
        fontSize: 32,
        fontWeight: 'bold'
      },
      sectionHeader: {
          fontSize: 16
      }
  });

export default PreviewContact;