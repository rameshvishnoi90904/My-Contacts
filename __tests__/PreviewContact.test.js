import { expect } from "@jest/globals";
import React from "react";
import renderer from "react-test-renderer";
import PreviewContact from '../src/screens/PreviewContact';
import { render } from "@testing-library/react-native";
import {contactList} from '../src/mockData/contactList';

describe("<PreviewContact />", () => {
    const mockData = contactList[0];

    it('renders correctly across screens', () => {
        const container = renderer.create(<PreviewContact route={{params: {contactInfo: mockData}}}/>)
        const tree = container.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('match contact name',() => {
        const component = render(<PreviewContact route={{params: {contactInfo: mockData}}}/>);
        const nameNode = component.getByText(`${mockData.givenName} ${mockData.familyName}`);
        expect(nameNode).toBeDefined();
    })


    it('match emailid',() => {
        const component = render(<PreviewContact route={{params: {contactInfo: mockData}}}/>);
        const emailNode = component.getByText(`${mockData.emailAddresses[0].email}`);
        expect(emailNode).toBeDefined();
    })

    it('match phone',() => {
        const component = render(<PreviewContact route={{params: {contactInfo: mockData}}}/>);
        const phoneNode = component.getByText(`${mockData.phoneNumbers[0].number}`);
        expect(phoneNode).toBeDefined();
    })
});