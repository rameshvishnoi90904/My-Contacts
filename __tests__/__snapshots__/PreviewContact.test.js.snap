// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviewContact /> renders correctly across screens 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginBottom": 16,
          "marginTop": 32,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "gray",
              "borderRadius": 60,
              "height": 120,
              "justifyContent": "center",
              "width": 120,
            },
            Object {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 32,
              "fontWeight": "bold",
            }
          }
        >
          EM
        </Text>
      </View>
    </View>
    <Text
      accessibilityRole="header"
      style={
        Object {
          "fontSize": 22,
          "textAlign": "center",
        }
      }
      testID="contact-name"
    >
      Elon
       
      Musk
    </Text>
    <View
      style={
        Object {
          "marginVertical": 8,
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#C9C9C9",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingBottom": 12,
            "paddingTop": 8,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
              }
            }
          >
            +23493023902
          </Text>
          <Text
            style={
              Object {
                "color": "#6D6D6D",
                "fontSize": 12,
                "textTransform": "capitalize",
              }
            }
          >
            personal
          </Text>
        </View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 22,
              },
              Object {
                "marginHorizontal": 16,
              },
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 8,
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#C9C9C9",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingBottom": 12,
            "paddingTop": 8,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            andre@gmail.com
          </Text>
          <Text
            style={
              Object {
                "color": "#6D6D6D",
                "fontSize": 12,
                "textTransform": "capitalize",
              }
            }
          >
            work
          </Text>
        </View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 22,
              },
              Object {
                "marginHorizontal": 16,
              },
              Object {
                "fontFamily": "Entypo",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "marginBottom": 16,
          "marginHorizontal": 16,
        }
      }
    >
      <View
        accessibilityLabel="Edit contact"
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Edit
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
